### PR TITLE
test: enable tests for `webpack-dev-server@beta-1`

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
tests

**Did you add tests for your changes?**
yes
**If relevant, did you update the documentation?**
No
**Summary**

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

**Comment by alexander-akait** [comment]
We need update snapshot resolver to include `webpack-dev-server` version too (otherwise snapshots will be broken), also some tests should be rewritten like `isDevServer3` and `isDevServer4`

**Review by alexander-akait — COMMENTED** [review]
I think we need snapshot resolver here, for webpack-dev-server v3 and webpack-dev-server v4, and for webpack v4 and v5

**Comment by snitin315** [comment]
Yep WIP.

**Comment by alexander-akait** [comment]
Ideally we should run it on the next `rc` release, I will do release and update snapshots, and all should work fine, thanks for helping

**Comment by alexander-akait** [comment]
I will finish it, anyway good job, big thanks

**Comment by alexander-akait** [comment]
Again, tests with changing directory and do not return to default state...

**Comment by alexander-akait** [comment]
@snitin315 Do you have idea where we have tests which change `process.cwd()`?

**Comment by snitin315** [comment]
No idea, from the logs it seems `loader` and `plugin` tests might be the culprits.

Screenshot 2021-04-08 at 3 23 12 PM
Screenshot 2021-04-08 at 3 23 28 PM

**Comment by anshumanv** [comment]
lets try to skip them and see if CI passes

**Comment by alexander-akait** [comment]
I can't understand which part of code change our cwd :disappointed:

**Comment by alexander-akait** [comment]
Found it is in `packages/generators/__tests__`

@webpack/cli-team We need rewrite these test on standard tests, somebody want to help?

**Comment by alexander-akait** [comment]
due `yeoman-test`

**Comment by alexander-akait** [comment]
I just drop them to test it here

File: package.json [hunk 1]
```diff
@@ -82,9 +82,9 @@
     "ts-jest": "^26.4.3",
     "ts-node": "^9.1.1",
     "typescript": "^4.1.3",
-    "webpack": "^5.25.0",
+    "webpack": "^4.46.0",
```
Comment (anshumanv) [inline package.json:85]: this shouldn't change

**Comment by anshumanv** [comment]
need to update some snapshots, I think it looks good now

**Comment by snitin315** [comment]
Same macOS issue 😞

**Comment by alexander-akait** [comment]
Merging is hell

**Comment by alexander-akait** [comment]
Hope all will be fine now and we can finish it, found some bug on webpack-dev-server side

**Pull request merged**